src/test/java/LinkedBinarySearchTreeTest.java

```diff
@@ -0,0 +1,832 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class LinkedBinarySearchTreeTest {
+
+    @Test
+    void isEmptyOnNewLinkedBinarySearchTreeReturnsTrue() {
```

Comment:
For all test names, `LinkedBinarySearchTree` is noise, you're already in the `LinkedBinarySearchTreeTest`, so of course all tests are related to the class under test. Here for example `isEmptyReturnsTrueForNewInstance` contains just as much information. Though, I do prefer `unitUnderTest_conditionUnderTest_expectedResult` test names, so `isEmpty_emptyTree_returnsTrue`.